EIPS/eip-1154.md

```diff
@@ -0,0 +1,94 @@
+---
+eip: 1154
+title: Oracle Interface
+author: Alan Lu (@cag)
+status: Draft
+type: Standards Track
+category: ERC
+created: 2018-06-13
+---
+
+## Simple Summary
+A standard interface for oracles.
+
+## Abstract
+In order for ethereum smart contracts to interact with off-chain systems, oracles must be used. These oracles report values which are normally off-chain, allowing smart contracts to react to the state of off-chain systems. A distinction and a choice is made between push and pull based oracle systems. Furthermore, a standard interface for oracles is described here, allowing different oracle implementations to be interchangeable.
+
+## Motivation
+The Ethereum ecosystem currently has many different oracle implementations available, but they do not provide a unified interface. Smart contract systems would be locked into a single set of oracle implementations, or they would require developers to write adapters/ports specific to the oracle system chosen in a given project.
```

Comment:
Can you provide example use-cases? What sort of oracles is this intended to support? Who would benefit from standardising such an interface?

Reply:
The use case I had in mind originally was for answering questions about "real-world events", where each ID can be correlated with a specification of a question and its answers (so most likely for prediction markets, basically).

Both the ID and the results are intentionally unstructured so that things like time series data (via splitting the ID) and different sorts of results (like one of a few, any subset of up to 256, or some value in a range with up to 256 bits of granularity) can be represented.

Another use case could be for decision-making processes, where the results given by the oracle represent decisions made by the oracle (e.g. futarchies).

Reply:
Can you expand on this in the EIP? And maybe make the title of the EIP more specific?